inloop/solutions/views.py

```diff
@@ -104,10 +106,20 @@ def get(self, request, **kwargs):
             junit.xml_to_dict(xml) for xml in xml_reports
         ]
 
+        upfiles = []
+        for solutionfile in solution.solutionfile_set.all():
+            with solutionfile.absolute_path as f:
+                upfiles.append({
+                    "code": f.open(encoding="utf-8").read(),
```

Comment:
Still wrong, the file handle returned by `f.open()` is never closed. You should use:
```python
with solutionfile.absolute_path.open(encoding="utf-8") as f:
    upfiles.append({
        "code": f.read(),
        # ...
    })
```